test/pages/me/day/[date].test.tsx

```diff
@@ -0,0 +1,16 @@
+import * as nextRouter from "next/router"
+import { shallow } from "enzyme"
+
+import Date from "../../../../src/pages/me/day/[date]"
+
+nextRouter.useRouter = jest.fn()
+
+describe("Me: ", () => {
+  it("matches snapshot", () => {
+    nextRouter.useRouter.mockImplementation(() => ({
+      query: { date: "01-01-2020" },
+    }))
+    const wrapper = shallow(<Date />)
+    expect(wrapper).toMatchSnapshot()
+  })
+})
```

Comment:
Should be able to get full coverage for this file by adding these two test cases (mount is needed because shallow doesn't call the useEffect hook):
```
  it("Renders nothing when date is not provided", () => {
    nextRouter.useRouter.mockImplementation(() => ({
      query: { date: null },
    }))
    const wrapper = shallow(<Date />)
    expect(wrapper).toMatchSnapshot()
  })

  it("Redirects to /me when no date is provided", () => {
    const mockPush = jest.fn()
    nextRouter.useRouter.mockImplementation(() => ({
      query: { date: null },
      push: mockPush,
    }))
    const wrapper = mount(<Date />)
    expect(mockPush).toHaveBeenCalledWith("/me")
  })
```

Reply:
I think this test also needs to be added for full coverage:
```
it("does not redirect to /me when date is provided", () => {
    const mockPush = jest.fn()
    nextRouter.useRouter.mockImplementation(() => ({
      query: { date: "01-01-2020" },
    }))
    mount(<Date />)
    expect(mockPush).toHaveBeenCalledTimes(0)
})
```